internal/backend/workloadproxy/reconciler.go

```diff
@@ -182,49 +156,50 @@ func (registry *Reconciler) GetProxy(alias string) (http.Handler, resource.ID, e
 		return nil, "", nil
 	}
 
-	lbSts := registry.clusterToAliasToLBStatus[clusterID][alias]
-	if lbSts == nil || lbSts.lb == nil {
+	upstreams := registry.clusterToAliasToBackend[clusterID][alias]
+	if upstreams == nil {
 		return nil, "", nil
 	}
 
-	hostPort := registry.hostPortForAlias(clusterID, alias)
-
-	targetURL := &url.URL{
-		Scheme: "http",
-		Host:   hostPort,
-	}
-
-	proxy := httputil.NewSingleHostReverseProxy(targetURL)
-	proxy.Transport = &http.Transport{
-		DialContext: registry.connProvider.DialContext,
-	}
+	return &httputil.ReverseProxy{
+		Director:  func(req *http.Request) { req.URL.Host = hostPortForAlias(clusterID, alias) },
+		Transport: registry.transport,
+	}, clusterID, nil
+}
 
-	return proxy, clusterID, nil
+// hostPortForAlias returns a unique IP:port for the given cluster and alias.
+func hostPortForAlias(clusterID resource.ID, alias string) string {
+	return fmt.Sprintf("%s_%s:4242", clusterID, alias)
```

Comment:
do we have some kind of guarantee that `_` can't be in the `clusterID` ?

Reply:
That's a good question. Maybe pick some other (perhaps unicode) symbol? But there is no guarantee that it will not be picked in the future either.There is one in `ClusterDiscoveryConfig.Validate` and one in `Config.Validate` checks in Talos, but both only ensure that ClusterID is not empty. As for Omni, I'm currently try to dig usages.

Reply:
use `net.JoinHostPort`